src/display/dom_utils.js

```diff
@@ -461,6 +461,72 @@ function isExternalLinkTargetSet() {
   }
 }
 
+class StatTimer {
+  constructor(enable = true) {
+    this.started = Object.create(null);
+    this.times = [];
+    this.enabled = !!enable;
+  }
+
+  time(name) {
+    if (!this.enabled) {
+      return;
+    }
+    if (name in this.started) {
+      warn('Timer is already running for ' + name);
+    }
+    this.started[name] = Date.now();
+  }
+
+  timeEnd(name) {
+    if (!this.enabled) {
+      return;
+    }
+    if (!(name in this.started)) {
+      warn('Timer has not been started for ' + name);
+    }
+    this.times.push({
+      'name': name,
+      'start': this.started[name],
+      'end': Date.now(),
+    });
+    // Remove timer from started so it can be called again.
+    delete this.started[name];
+  }
+
+  toString() {
+    let times = this.times;
+    // Find the longest name for padding purposes.
+    let out = '', longest = 0;
+    for (let i = 0, ii = times.length; i < ii; ++i) {
+      let name = times[i]['name'];
+      if (name.length > longest) {
+        longest = name.length;
+      }
+    }
+    for (let i = 0, ii = times.length; i < ii; ++i) {
+      let span = times[i];
+      let duration = span.end - span.start;
+      out += `${span['name'].padEnd(longest)} ${duration}ms\n`;
+    }
+    return out;
+  }
+}
+
+class DummyStatTimer {
```

Comment:
Could you add a small comment to this class as to why it exists? I think you described it pretty well in the pull request description, but let's add it here too for reference.

Reply:
Sure, I've tried to add a comment; please check if it makes sense!

Reply:
Yes, that's exactly how I thought it should look. Nice!